src/basic/fileio.c

```diff
@@ -148,6 +148,28 @@ int write_string_stream_ts(
                         return -EBADF;
         }
 
+        if (flags & WRITE_STRING_VIRTUALFILE_SUPRESS_IF_POSSIBLE) {
+                _cleanup_free_ char *ret = NULL;
```

Comment:
please use a different name here. We use ret/ret_xyz as name for return parameters of the current funciton, but here you just want to pass a variable to some other call, i.e. from the PoV of this function it's not a return parameter.